packages/core/notifications/core-notifications-browser-internal/src/toasts/global_toast_list.tsx

```diff
@@ -31,50 +34,77 @@ const convertToEui = (toast: ToastWithRichTitle): EuiToast => ({
   text: toast.text instanceof Function ? <MountWrapper mount={toast.text} /> : toast.text,
 });
 
-export class GlobalToastList extends React.Component<Props, State> {
-  public state: State = {
-    toasts: [],
-    idToToasts: {},
-  };
+export const GlobalToastList: FunctionComponent<Props> = ({
+  toasts$,
+  dismissToast,
+  reportEvent,
+}) => {
+  const [toasts, setToasts] = useState<State['toasts']>([]);
+  const [idToToasts, setIdToToasts] = useState<State['idToToasts']>({});
 
-  private subscription?: Subscription;
+  const reportToastDismissal = useCallback(
+    (representedToasts: State['idToToasts'][number]) => {
+      // Select the first duplicate toast within the represented toast group
+      // given it's identical to all other recurring ones within it's group
+      const firstDuplicateToast = representedToasts[0];
 
-  public componentDidMount() {
-    this.subscription = this.props.toasts$.subscribe((redundantToastList) => {
-      const { toasts, idToToasts } = deduplicateToasts(redundantToastList);
-      this.setState({ toasts, idToToasts });
+      if (
+        representedToasts.length > 1 &&
+        firstDuplicateToast.color !== 'success' &&
+        firstDuplicateToast.title
+      ) {
+        reportEvent.onDismissToast({
+          toastMessage:
+            firstDuplicateToast.title instanceof Function
+              ? renderToStaticMarkup(<MountWrapper mount={firstDuplicateToast.title} />)
+              : firstDuplicateToast.title,
```

Comment:
There is risk that we are sending [PII](https://en.wikipedia.org/wiki/Personal_data) as part of this telemetry event. I'm not sure what the most practical way is to mitigate this...

1. Have an opaque ID for sources of toasts (most effort)
2. Maybe use stack traces? Haven't tested if it'd work and will be super noisey
3. Calculate a representation of the title string (like a hash?). probably not useful in end result though...
4. What else?

Reply:
Thanks, for bringing this up, actually discussing this same thing in core Slack channe.

Reply:
We will remove the title.

Reply:
Actually, a U-turn, we were informed in Slack that collecting title is OK.